app/src/pages/inside/launchesPage/modals/deleteLaunchModal/deleteLaunchModal.jsx

```diff
@@ -0,0 +1,245 @@
+  },
+  deleteImportantLaunchMessage: {
+    id: 'DeleteLaunchModal.deleteImportantLaunchMessage',
+    defaultMessage:
+      '<b>”{name}” is marked as important.</b><br />Are you sure you want to delete this important launch?',
+  },
+  deleteImportantLaunchesMessage: {
+    id: 'DeleteLaunchModal.deleteImportantLaunchesMessage',
+    defaultMessage:
+      '<b>There are {importantCount} launches marked as important.</b><br />Are you sure you want to delete these important launches?',
+  },
+});
+
+const cx = classNames.bind(styles);
+
+const DeleteLaunchModal = ({ data: { launches, confirmDeleteLaunches, userId } }) => {
+  const { formatMessage } = useIntl();
+  const confirmAndClose = (closeModal) => {
+    confirmDeleteLaunches(launches);
+    closeModal();
+  };
+
+  const okButton = {
+    text: formatMessage(COMMON_LOCALE_KEYS.DELETE),
+    danger: true,
+    onClick: confirmAndClose,
+  };
+  const cancelButton = {
+    text: formatMessage(COMMON_LOCALE_KEYS.CANCEL),
+  };
+
+  const selectedImportantLaunches = launches.filter(
+    (launch) => launch.retentionPolicy === RETENTION_POLICY.IMPORTANT,
+  );
+
+  const selectedRegularLaunches = launches.filter(
+    (launch) => launch.retentionPolicy === RETENTION_POLICY.REGULAR,
+  );
+
+  const getMainContent = () => {
+    if (selectedImportantLaunches.length > 0 && selectedRegularLaunches.length > 0) {
+      return formatMessage(
+        selectedImportantLaunches.length === 1
+          ? messages.deleteWithImportantLaunchMessage
+          : messages.deleteWithImportantLaunchesMessage,
+        {
+          b: (data) => DOMPurify.sanitize(`<b>${data}</b>`),
+          name: selectedImportantLaunches[0].name,
+          importantCount: selectedImportantLaunches.length,
+          totalCount: launches.length,
+        },
+      );
+    } else if (selectedImportantLaunches.length > 0) {
+      return formatMessage(
+        selectedImportantLaunches.length === 1
+          ? messages.deleteImportantLaunchMessage
+          : messages.deleteImportantLaunchesMessage,
+        {
+          b: (data) => DOMPurify.sanitize(`<b>${data}</b>`),
+          name: selectedImportantLaunches[0].name,
+          importantCount: selectedImportantLaunches.length,
+        },
+      );
+    } else {
+      return launches.length === 1
+        ? formatMessage(messages.deleteModalContent, {
+            b: (data) => DOMPurify.sanitize(`<b>${data}</b>`),
+            name: launches[0].name,
+          })
+        : formatMessage(messages.deleteModalMultipleContent);
+    }
+  };
+
+  const isNotAllOwnLaunches = launches.some((launch) => launch.owner !== userId);
+
+  const warning =
+    isNotAllOwnLaunches &&
+    (launches.length === 1
+      ? formatMessage(messages.warning)
+      : formatMessage(messages.warningMultiple));
+
+  const Footer = ({ onClickOk, closeHandler }) => (
```

Comment:
Move the component definition outside the Modal component and handle dynamic usage via `useCallback`,